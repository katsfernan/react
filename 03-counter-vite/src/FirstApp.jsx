// import { Fragment } from 'react'
import PropTypes from 'prop-types';

const newMessage = {
    message: 'Hola mundo',
    title: 'Fernando'
}

const convierteMessage = ({message, title}) => `message ${message} and author: ${title}`

export const FirstApp = ({ title, subTitle }) => {

    return (
        // <Fragment>
        //     <h1>Hola Mundo</h1>
        //     <p>Soy un subtitulo</p>
        // </Fragment>
        <>
            <h1 data-testid = 'test-title'>{ title }</h1>
            <h2>{ subTitle }</h2>
        </>
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired

}

FirstApp.defaultProps = {
    // title: 'No hay ningun titulo',
    subTitle: 'No hay ningun subtitulo'
}
