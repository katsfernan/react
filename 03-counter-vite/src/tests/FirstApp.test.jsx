import { render } from '@testing-library/react'
import { FirstApp } from '../FirstApp';

describe('Pruebas en <FirstApp/>', () => { 
    
    test('debe de hacer match con el snapshot', () => { 
    
        const { container } = render(
        <FirstApp 
            title='Hola, soy Fernando'
            subTitle = 'Soy un subtitulo'
        /> 
         )

        expect(container).toMatchSnapshot();

     });
    
    test('debe de mostrar el titulo en un h1', () => { 

        const title = 'Hola, que tal todo ?'
        const { container, getByText } = render(
        <FirstApp 
            title={title}
            subTitle = 'Soy un subtitulo'
        /> 
         )

         expect( getByText(title) ).toBeTruthy();

         const h1 = container.querySelector('h1')

         expect(h1.innerHTML).toBe(title)
     })
 })