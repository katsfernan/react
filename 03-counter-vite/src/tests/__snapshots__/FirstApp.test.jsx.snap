// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Fernando
  </h1>
  <h1>
    Soy un subtitulo
  </h1>
</div>
`;
